pkg/controllers/v1alpha1/datatable/datatable_controller.go

```diff
@@ -103,9 +114,63 @@ func (r *DataTableReconciler) reconcileDataTable(ctx reconcileRequestContext) (r
 		return utils.RequeueAfterInterval(10 * time.Second)
 	}
 
+	// 1. create the hive client
+	host := ctx.Datatable.Spec.Url
+	conn, err := hive.CreateHiveClient(host)
```

Comment:
Seems that the `conn` variable will not be closed except for `r.ReconcileDatatableDeletion`. I think this may lead to memory issues.

Reply:
yes, i will fix it